src/components/security_manager/src/security_manager_impl.cc

```diff
@@ -565,6 +566,29 @@ bool SecurityManagerImpl::ProcessHandshakeData(
   } else if (handshake_result != SSLContext::Handshake_Result_Success) {
     // On handshake fail
     SDL_LOG_WARN("SSL initialization finished with fail.");
+    int32_t error_code = ERROR_HANDSHAKE_FAILED;
+    std::string error_text = "Handshake failed";
+    switch (handshake_result) {
+      case SSLContext::Handshake_Result_CertExpired:
+        error_code = ERROR_EXPIRED_CERT;
+        error_text = "Certificate is expired";
+        break;
+      case SSLContext::Handshake_Result_NotYetValid:
+        error_code = ERROR_EXPIRED_CERT;
```

Comment:
I think a different error code should be returned for this one. Maybe just the default or ERROR_INVALID_CERT since the cert will eventually be valid. It is not expired so this could cause some confusion for people debugging in this case.
```suggestion
        error_code = ERROR_INVALID_CERT;
```